Sync the mitxonline Zendesk theme with the upstream. (#2)

marslanabdulrauf opened the pull request:
* style: zendesk Theme updates

* style: zenDesk theme updates: round 1

* style: zendesk round 2 changes

* fix: removed phone number from footer

* fix: update the font sizes

* style: improve readability of Support Center (Zendesk) pages

* feat: add Google Tag Manager to Zendesk pages (#8)

* add Google Tag Manager to Zendesk pages

* address Brian's feedback

* add empty line at the end of file

* fix: update copyright date (#9)

* update copyright date

* fixes

* fix: update favicon (#10)

* fix: update document_head.hbs

Added Montserrat font

* fix: update manifest.json

Switched to Montserrat font

* fix: update manifest.json

Updated default settings

* fix: delete logo.png

deleted logo

* fix: add files via upload

Added logo

* fix: delete logo.png

* fix: add files via upload

* fix: update manifest.json

xpro => mitxoline

* fix: update en-us.json

xPRO => mitxonline

* fix: update header.hbs

xpro => mitxonline

* fix: update footer.hbs

xpro => mitxonline

* fix: update _header.scss

set up for mitxonline

* fix: update style.css

set up for mitxonline

* fix: update style.css

logo width

* fix: update _header.scss

logo width

* style: update footer design to match MITxOnline's footer (#4)

* style: tweaks for request, faq and other pages (#6)

* fix: search input text/placeholder color (#7)

* fix: update input fields text color

* style: update header footer to match marketing site (#1)

* style: logo update and header/footer match MITx Online
* update readme
* Update fonts

* fix: upstream changes (part 1)

---------


## Additional Context
https://developer.zendesk.com/api-reference/help_center/help-center-templates/new_request_page/#available-helpers

`follow_up` is removed in v4
`request_form` helper is replaced with `new_request_form` object which can be used to render custom forms
`new-request-form` is shifted to React component, Regular/previous classes might not work

## Testing URL: https://arbisoft7137.zendesk.com

arslanashraf7 commented:
@briangrossman can you take a look at the screenshots and/or the test theme link that i added in the PR description. let me know if this looks good to you.

briangrossman commented:
@arslanashraf7 
LGTM - where G = Great

One question: Why are there two `Subject` fields in the [Submit a Request](https://mitxonlinesupporttest.zendesk.com/hc/en-us/requests/new) page? 
![image](https://user-images.githubusercontent.com/21342568/147979513-5155768c-3022-4149-a823-8419249a5d9d.png)

arslanashraf7 commented:
> @arslanashraf7 LGTM - where G = Great
> 
> One question: Why are there two `Subject` fields in the [Submit a Request](https://mitxonlinesupporttest.zendesk.com/hc/en-us/requests/new) page? ![image](https://user-images.githubusercontent.com/21342568/147979513-5155768c-3022-4149-a823-8419249a5d9d.png)

@briangrossman, I created an extra field in the form of inputs through Zendesk admin in [this](https://mitxonlinesupporttest.zendesk.com/hc/en-us) test deployment (Which I've now removed from the test account as well). This won't affect things in production.

umarmughal824 reviewed (approved):
:+1:

arslanashraf7 commented:
@briangrossman these changes are merged and you can deploy the latest theme from the main branch. Let me know if you see any other issues.

mudassir-hafeez committed:
Sync the mitxonline Zendesk theme with the upstream. (#2)

* style: zendesk Theme updates

* style: zenDesk theme updates: round 1

* style: zendesk round 2 changes

* fix: removed phone number from footer

* fix: update the font sizes

* style: improve readability of Support Center (Zendesk) pages

* feat: add Google Tag Manager to Zendesk pages (#8)

* add Google Tag Manager to Zendesk pages

* address Brian's feedback

* add empty line at the end of file

* fix: update copyright date (#9)

* update copyright date

* fixes

* fix: update favicon (#10)

* fix: update document_head.hbs

Added Montserrat font

* fix: update manifest.json

Switched to Montserrat font

* fix: update manifest.json

Updated default settings

* fix: delete logo.png

deleted logo

* fix: add files via upload

Added logo

* fix: delete logo.png

* fix: add files via upload

* fix: update manifest.json

xpro => mitxoline

* fix: update en-us.json

xPRO => mitxonline

* fix: update header.hbs

xpro => mitxonline

* fix: update footer.hbs

xpro => mitxonline

* fix: update _header.scss

set up for mitxonline

* fix: update style.css

set up for mitxonline

* fix: update style.css

logo width

* fix: update _header.scss

logo width

* style: update footer design to match MITxOnline's footer (#4)

* style: tweaks for request, faq and other pages (#6)

* fix: search input text/placeholder color (#7)

* fix: update input fields text color

* style: update header footer to match marketing site (#1)

* style: logo update and header/footer match MITx Online
* update readme
* Update fonts

* fix: upstream changes (part 1)

---------

Co-authored-by: asadiqbal <aiqbal@edx.org>
Co-authored-by: Umar Asghar <umar.asghar@arbisoft.com>
Co-authored-by: Brian Grossman <bgrossman@alum.mit.edu>
Co-authored-by: Arslan Ashraf <34372316+arslanashraf7@users.noreply.github.com>
Co-authored-by: Anna Gavrilman <annagav@mit.edu>

pdpinch commented:
The search button isn't styled correctly. 

Compare https://arbisoft7137.zendesk.com/hc/en-us

![image](https://github.com/user-attachments/assets/25335a99-98ee-4541-b94c-57be9c899601)

to https://mitxonline.zendesk.com/hc/en-us

![image](https://github.com/user-attachments/assets/acdbd664-07f2-4053-b4a2-2dd2c3e9778e)

Most everything else looks ok, although the contact form fields on https://arbisoft7137.zendesk.com/hc/en-us/requests/new seems to be configured very differently from https://mitxonline.zendesk.com/hc/en-us/requests/new -- I'm not too concerned about the styling of the form though.

arslanashraf7 commented:
@pdpinch I noticed the same (+ 1 additional) things during my testing so far:

 1) The search button 2) The form fields styling 3) The new request submit button styling

<img width="718" alt="image" src="https://github.com/user-attachments/assets/261ca6e2-0529-4664-b61a-8f074569612c" />

arslanashraf7 commented:
@marslanabdulrauf the search button design is fixed but the Submit button & form design are still inconsistent between our live theme and the test theme against this PR:

**Live Theme (Form)**
<img width="859" alt="image" src="https://github.com/user-attachments/assets/c5644c6b-f981-404c-a023-a14e017eb816" />

**Theme from this branch (Form)**
<img width="865" alt="image" src="https://github.com/user-attachments/assets/8b78209f-59e2-4c1e-a494-e4867e7cf56b" />

**Live Theme (Submit button )**
<img width="819" alt="image" src="https://github.com/user-attachments/assets/955eedfa-4ed8-495b-b114-0a9569214eb2" />

**Theme from this branch (Submit Button)**

<img width="854" alt="image" src="https://github.com/user-attachments/assets/039eba05-fd7f-4570-be78-8c45d96eaa82" />

marslanabdulrauf commented:
@pdpinch The search button issue has been fixed.

@arslanashraf7 The `Submit` button placement is also corrected. However, since the `new-request-form` has been migrated to a React component, we have lost direct control over its styling. React applies unique, auto-generated class names, which makes overriding styles more challenging.

Please refer to the test link for further verification.

marslanabdulrauf committed:
fix: button styling fixed on home and new request page

arslanashraf7 reviewed (approved):
Thanks for making all the changes, This looks good. 👍 

There is still a slight font size difference between https://mitxonline.zendesk.com/hc/en-us and https://arbisoft7137.zendesk.com/hc/en-us but that's neglectable for me.

Peter was already ok with new request form design so it's a 👍 there from my side too.

Don't forget to squash merge when you do.
